aws-amplify/docs
# Add script to generate directory object

#### Description of changes:
Staging site: https://next-release-directory-script.d3kzzkiugubm89.amplifyapp.com/

- Created `generateDirectory.mjs`
  - It traverses the directory object from `directory.mjs` and uses that to find the respective markdown files containing the `meta` object. Then it copies the data from there onto a copy of the directory object and saves it to a new json file
  - By traversing the directory object from `directory.mjs` we're able to keep the order of how things should be displayed in components like the Menu

To test:
1. Pull down the branch
2. Run a `yarn && yarn dev`
3. Inside the `directory/` for your source code, you should have an updated `directory.json` of the generated file

#### Related GitHub issue #, if available:

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?

Which platform(s) are affected by this PR (if applicable)?

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks




      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._
